pkgs/tools/graphics/wkhtmltopdf-bin/default.nix

```diff
@@ -1,28 +1,81 @@
-{ stdenv, lib, fetchurl, wkhtmltopdf, xar, cpio }:
+{ autoPatchelfHook
+, cpio
+, freetype
+, zlib
+, openssl
+, dpkg
+, fetchurl
+, gcc-unwrapped
+, lib
+, libjpeg8
+, libpng
+, fontconfig
+, stdenv
+, wkhtmltopdf
+, xar
+, xorg
+}:
 
-stdenv.mkDerivation rec {
+let
+  inherit (lib)
+    getLib
+    optionalAttrs
+    ;
 
-  pname = "wkhtmltopdf-bin";
-  version = "0.12.6-1";
-  sha256 = "1db59kdprzpmvdj1bg47lmfgi3zlvzvqif11sbym9hw61xy6gp3d";
-  src = fetchurl {
-    url =
-      "https://github.com/wkhtmltopdf/packaging/releases/download/${version}/wkhtmltox-${version}.macos-cocoa.pkg";
-    inherit sha256;
+  darwinAttrs = rec {
+    version = "0.12.6-2";
+    src = fetchurl {
+      url = "https://github.com/wkhtmltopdf/packaging/releases/download/${version}/wkhtmltox-${version}.macos-cocoa.pkg";
+      sha256 = "sha256-gaZrd7UI/t6NvKpnEnIDdIN2Vos2c6F/ZhG21R6YlPg=";
+    };
+
+    nativeBuildInputs = [ xar cpio ];
+
+    unpackPhase = ''
+      xar -xf $src
+      zcat Payload | cpio -i
+      tar -xf usr/local/share/wkhtmltox-installer/wkhtmltox.tar.gz
+    '';
+
+    installPhase = ''
+      mkdir -p $out
+      cp -r bin include lib share $out/
+    '';
   };
 
-  buildInputs = [ xar cpio ];
+  linuxAttrs = rec {
+    version = "0.12.6-3";
+    src = fetchurl {
+      url = "https://github.com/wkhtmltopdf/packaging/releases/download/${version}/wkhtmltox-${version}.archlinux-x86_64.pkg.tar.xz";
+      sha256 = "sha256-6Ewu8sPRbqvYWj27mBlQYpEN+mb+vKT46ljrdEUxckI=";
+    };
 
-  unpackPhase = ''
-    xar -xf $src
-    zcat Payload | cpio -i
-    tar -xf usr/local/share/wkhtmltox-installer/wkhtmltox.tar.gz
-  '';
+    nativeBuildInputs = [ autoPatchelfHook ];
 
-  installPhase = ''
-    mkdir -p $out
-    cp -r bin include lib share $out/
-  '';
+    buildInputs = [
+      xorg.libXext
+      xorg.libXrender
+
+      freetype
+      openssl
+      zlib
+
+      (getLib fontconfig)
+      (getLib gcc-unwrapped)
+      (getLib libjpeg8)
+      (getLib libpng)
+    ];
+
+    unpackPhase = "tar -xf $src";
+
+    installPhase = ''
```

Comment:
`runHook` for preInstall and postInstall is missing. Maybe it would be easier to define a list of directories here and use a common `installPhase` in `mkDerivation`?

Reply:
Can you elaborate on the list of directories? I'm not sure I've seen that before. I don't see anything about it in the docs either: https://nixos.org/manual/nixpkgs/unstable/#ssec-install-phase

Reply:
I meant an approach that's not built into mkDerivation. Something like:
```
let darwinAttrs.... linuxAttrs....
  installDirs = for darwin "bin include lib share", for linux "usr/bin usr/include usr/lib usr/share";
in
stdenv.mkDerivation ({
...
  installPhase = ''
    runHook preInstall
    mkdir -p $out
    cp -r ${installDirs} $out/
    runHook postInstall
  '';
```
I don't know if that's better than repeating the whole installPhase twice... just something to think about.

Reply:
I'm not sure that helps, but I'll go ahead and add the pre/post hooks.